TestCustomConfiguration is flaky

```
=== RUN   TestCustomConfiguration
--- FAIL: TestCustomConfiguration (0.00s)
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 52855 [running]:
testing.tRunner.func1(0xc0012e6000)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x166ce20, 0x1ab6030)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch.Builder.WithPodTemplate(0x0, 0x0, 0x0, 0x0, 0xc0014c9180, 0xd, 0x0, 0x0, 0xc0008e85a0, 0x11, ...)
	/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch/builder.go:279 +0xd4
github.com/elastic/cloud-on-k8s/test/e2e/es.TestCustomConfiguration(0xc0012e6000)
	/go/src/github.com/elastic/cloud-on-k8s/test/e2e/es/podtemplate_test.go:37 +0x49d
testing.tRunner(0xc0012e6000, 0x19058a8)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350
FAIL	github.com/elastic/cloud-on-k8s/test/e2e/es	5652.649s
```

--- a/test/e2e/test/elasticsearch/builder.go
+++ b/test/e2e/test/elasticsearch/builder.go
@@ -275,8 +275,11 @@ func (b Builder) WithDefaultPersistentVolumes() Builder {
 }
 
 func (b Builder) WithPodTemplate(pt corev1.PodTemplateSpec) Builder {
+	if pt.Labels == nil {
+		pt.Labels = make(map[string]string)
+	}
+	pt.Labels[run.TestNameLabel] = b.Elasticsearch.Labels[run.TestNameLabel]
 	for i := range b.Elasticsearch.Spec.NodeSets {
-		pt.Labels[run.TestNameLabel] = b.Elasticsearch.Labels[run.TestNameLabel]
 		b.Elasticsearch.Spec.NodeSets[i].PodTemplate = pt
 	}
 	return b

